src/components/file-browser/FileBrowserDirectories.ts

```diff
@@ -0,0 +1,369 @@
+        [isExpanded && !hasLoadedContents, () => 'loadingSpinner'],
+        [!isExpanded, () => 'angle-right']
+      ), {
+        // @ts-expect-error
+        color: isSelected ? '#000' : colors.accent(),
+        size: 14
+      })
+    ]),
+    h(Link, {
+      id: `${id}-link`,
+      role: 'presentation',
+      tabIndex: -1,
+      style: {
+        display: 'inline-block',
+        overflow: 'hidden',
+        maxWidth: '100%',
+        padding: `0.25rem 0.5rem 0.25rem ${level + 1.25}rem`,
+        borderColor: id === activeDescendant ? colors.accent() : 'transparent',
+        borderStyle: 'solid',
+        borderWidth: '1px 0',
+        textOverflow: 'ellipsis',
+        whiteSpace: 'nowrap',
+        ...(isSelected && {
+          color: '#000'
+        })
+      },
+      ...(isSelected && {
+        hover: { color: '#000' }
+      }),
+      onClick: () => {
+        onSelectDirectory(path)
+      }
+    }, [path === '' ? rootLabel : basename(path)]),
+    isExpanded && h(FileBrowserDirectoryContents, {
+      activeDescendant,
+      level,
+      parentId: id,
+      path,
+      provider,
+      rootLabel,
+      selectedDirectory,
+      setActiveDescendant,
+      onFinishedLoading: () => setHasLoadedContents(true),
+      onSelectDirectory
+    })
+  ])
+}
+
+interface FileBrowserDirectoriesProps {
+  provider: FileBrowserProvider
+  selectedDirectory: string
+  onSelectDirectory: (path: string) => void
+}
+
+const FileBrowserDirectories = (props: FileBrowserDirectoriesProps) => {
+  const {
+    provider,
+    selectedDirectory,
+    onSelectDirectory
+  } = props
+
+  const treeElementRef = useRef<HTMLUListElement | null>(null)
+
+  const [activeDescendant, setActiveDescendant] = useState('node-0')
+
+  return h(Interactive, {
+    as: 'ul',
+    ref: treeElementRef,
+    // aria-activedescendant tells which tree item is "focused", while actual focus stays on the tree itself.
+    'aria-activedescendant': activeDescendant,
+    'aria-label': 'Workspace files',
+    role: 'tree',
+    tabIndex: 0,
+    style: {
+      padding: 0,
+      border: '2px solid transparent',
+      margin: 0,
+      listStyleType: 'none'
+    },
+    hover: {
+      border: `2px solid ${colors.accent()}`
+    },
+    onKeyDown: e => {
+      // If the key isn't relevant to tree navigation, do nothing.
+      if (!(e.key === 'Enter' || e.key.startsWith('Arrow'))) {
+        return
+      }
+
+      e.preventDefault()
+      e.stopPropagation()
+
+      const currentTreeItem = document.getElementById(activeDescendant)
+
+      if (!currentTreeItem) {
+        // If the active descendant isn't found (for example, if it was in a group that has been collapsed),
+        // then reset the active descendant to the first item in the tree.
+        setActiveDescendant('node-0')
+      } else if (e.key === 'Enter') {
+        if (currentTreeItem.id.endsWith('-load-next')) {
+          // If on a load next page tree item, load the next page.
```

Comment:
It would be nice if the keyboard focus went to the first loaded node when "Load Next Page" completes. As it is, you have to navigate back up and understand what was there before you triggered the action.

Reply:
Added in https://github.com/DataBiosphere/terra-ui/pull/3469/commits/2b09e10ee894418346ff07931134e02ba78f8a45.